NetTally/ReorderTasksWindow.xaml.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using NetTally.Utility;
+using NetTally.ViewModels;
+
+namespace NetTally
+{
+    /// <summary>
+    /// Interaction logic for ReorderTasks.xaml
+    /// </summary>
+    public partial class ReorderTasksWindow : Window, INotifyPropertyChanged
+    {
+        #region Constructor and variables
+        public ListCollectionView TaskView { get; }
+
+        MainViewModel MainViewModel { get; }
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        public ReorderTasksWindow()
+        {
+            InitializeComponent();
+        }
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mainViewModel">The primary view model of the program</param>
+        public ReorderTasksWindow(MainViewModel mainViewModel)
+        {
+            InitializeComponent();
+
+            MainViewModel = mainViewModel;
+
+            MainViewModel.PropertyChanged += MainViewModel_PropertyChanged;
+
+            // Create filtered, sortable views into the collection for display in the window.
+            if (MainViewModel.TaskList != null)
+            {
+                TaskView = new ListCollectionView(MainViewModel.TaskList);
+            }
+
+            // Initialize starting selected positions
+            TaskView.MoveCurrentToPosition(-1);
+
+            // Set the data context for binding.
+            DataContext = this;
+
+            // Update the displayed order immediately
+            UpdateTaskList();
+        }
+        #endregion
+
+        private void UpdateTaskList()
+        {
+            VoteCounter.Instance.OrderedTaskList = TaskView.SourceCollection as List<string>;
+            OnPropertyChanged("Votes");
```

Comment:
Afraid I'm not sure what you mean here. I wracked my brains for ages trying to work out how to give VoteCounter access to the ordered list from the ReorderTasksWindow, and this is what I came up with. VoteCounter doesn't get passed MainViewModel anywhere, so it can't easily just look at MainViewModel.TaskList.

I'm afraid my grasp of events is a little shaky, I'm not seeing how to use one to pass the required information.